lib/util/addExtraParamsTo.js

```diff
@@ -10,7 +10,11 @@ function addExtraParamsTo(routeArguments) {
     //  is null. WWARNER 08/21/2014
     paramsAndRequest.pop();
 
-    var extraParams = Array.prototype.slice.call(arguments, 1);
```

Comment:
Still not sure I understand after reading that. In any case, follow their example of using the array constructor to initialize its size upfront. Then index in rather than pushing.